packages/plugin-kit/src/fs/readProjectFile.ts

```diff
@@ -0,0 +1,24 @@
+import * as fs from "node:fs/promises";
+
+import { SliceMachineHelpers } from "../createSliceMachineHelpers";
+
+type BufferEncoding = Extract<Parameters<typeof fs.readFile>[1], string>;
+
+export type ReadProjectFileArgs = {
+	filename: string;
+	helpers: SliceMachineHelpers;
+};
+
+export async function readProjectFile(
+	args: ReadProjectFileArgs,
+): Promise<Buffer>;
+export async function readProjectFile(
+	args: ReadProjectFileArgs & { encoding: BufferEncoding },
+): Promise<string>;
+export async function readProjectFile(
+	args: ReadProjectFileArgs & { encoding?: BufferEncoding },
+): Promise<Buffer | string> {
```

Comment:
Any thoughts on using the arrow function overload syntax instead for those helpers?

```typescript
export const readProfileFile: {
	(args: ReadProjectFileArgs): Promise<Buffer>;
	(args: ReadProjectFileArgs & { encoding: BufferEncoding }): Promise<string>;
} = async(
	args: ReadProjectFileArgs & { encoding?: BufferEncoding },
): Promise<Buffer | string> {}
```

We did so on client helpers: https://github.com/prismicio/prismic-client/blob/76fa2ca4e207d70fa3a19f131d9a416a9dfb180a/src/helpers/asText.ts

Reply:
I gave it a try and ran into type issues with the return type. It doesn't seem to be an issue with `asText()` since the return types are the same across each overload.

Reply:
Alright, no worries, works for me :)